compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -464,7 +464,26 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                                 root_obligation,
                             )
                         } else {
-                            (trait_predicate, &obligation)
+                            let mut temp_trait_predicate = trait_predicate;
+                            let mut loop_code = obligation.cause.code().clone();
+                            loop{
```

Comment:
Can you run `./x.py fmt`? This might not be the only thing that CI will complain about.

```suggestion
                            loop {
```

Reply:
It's probably that rustfmt bailed in this method because of something else, like a string that's too long or a where clause somewhere that it can't handle, because otherwise tidy would have failed.

Reply:
Probably the `matches!`